FreeRTOS/FreeRTOS-Kernel
# Declare vApplicationMallocFailedHook function in task.h instead in each C heap file

Declare vApplicationMallocFailedHook function in task.h instead in each C heap file

Description
-----------
vApplicationMallocFailedHook  was declared in each Heap file. which forces users to declare it and can cause problems if the prototype of the function changes.
So I done like other hook, example vApplicationStackOverflowHook.

Test Steps
-----------
Compile with GCC on my project

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
